### PR TITLE
Fix how to use url_name parameter in features and attributes

### PR DESCRIPTION
**Pull request by PierreRambaud**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Description?  | Use the url_name when it's defined..
| Type?         | improvement
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | Fixes PrestaShop/Prestashop#16939
| How to test?  | See ticket.

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/ps_facetedsearch/301)

### CODE REVIEW

File: src/Filters/DataAccessor.php [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+        if (!Combination::isFeatureActive()) {
+            return [];
+        }
+
+        if (!isset($this->attributes[$idLang][$idAttributeGroup])) {
+            $this->attributes[$idLang] = [$idAttributeGroup => []];
+            $tempAttributes = $this->database->executeS(
+                'SELECT DISTINCT a.`id_attribute`, ' .
+                'a.`color`,  ' .
+                'al.`name`,  ' .
```
Comment (PierreRambaud) [inline src/Filters/DataAccessor.php:68]:
```suggestion
                'a.`color`, ' .
                'al.`name`, ' .
```

File: tests/php/FacetedSearch/Filters/BlockTest.php [hunk 2]
```diff
@@ -680,17 +694,21 @@ public function testGetFiltersBlockWithAttributesWithAttributesGroupsWithoutFeat
...
+                'SELECT DISTINCT a.`id_attribute`, a.`color`,  al.`name`,  agl.`id_attribute_group`, IF(lialv.`url_name` IS NULL OR lialv.`url_name` = "", NULL, lialv.`url_name`) AS url_name, IF(lialv.`meta_title` IS NULL OR lialv.`meta_title` = "", NULL, lialv.`meta_title`) AS meta_title FROM `ps_attribute_group` ag INNER JOIN `ps_attribute_group_lang` agl ON (ag.`id_attribute_group` = agl.`id_attribute_group` AND agl.`id_lang` = 2) INNER JOIN `ps_attribute` a ON a.`id_attribute_group` = ag.`id_attribute_group` INNER JOIN `ps_attribute_lang` al ON (a.`id_attribute` = al.`id_attribute` AND al.`id_lang` = 2)INNER JOIN ps_attribute_group_shop attribute_group_shop ON (attribute_group_shop.id_attribute_group = ag.id_attribute_group AND attribute_group_shop.id_shop = 1) INNER JOIN ps_attribute_shop attribute_shop ON (attribute_shop.id_attribute = a.id_attribute AND attribute_shop.id_shop = 1) LEFT JOIN `ps_layered_indexable_attribute_lang_value` lialv ON (a.`id_attribute` = lialv.`id_attribute` AND lialv.`id_lang` = 2) WHERE ag.id_attribute_group = 1 ORDER BY agl.`name` ASC, a.`position` ASC'
```
[the remark this answers is not in the document]
Comment (PierreRambaud) [reply tests/php/FacetedSearch/Filters/BlockTest.php]: Thanks, I had to change the linked code too but it should be ok right now

**Review by atomiix — APPROVED** [review]

**Comment by khouloudbelguith** [comment]
Hi @PierreRambaud,

I tried to use this PR with PS1771 and with develop branch, but the same issue is displayed
**BO**
image

**FO**
image


- I tried with php7.1 & PHP7.2.
- I regenerated the .htaccess
- I used an incognito web 

What do you think?

Thanks!

**Comment by PierreRambaud** [comment]
> Hi @PierreRambaud,
> 
> I tried to use this PR with PS1771 and with develop branch, but the same issue is displayed
> **BO**
> image
> 
> **FO**
> image
> 
> * I tried with php7.1 & PHP7.2.
> * I regenerated the .htaccess
> * I used an incognito web
> 
> What do you think?
> 
> Thanks!

As I said here https://github.com/PrestaShop/PrestaShop/issues/16939#issuecomment-752125462 this module never aims to change the product page behaviour.

**Comment by khouloudbelguith** [comment]
Hi @PierreRambaud,

I checked with PS1771 & the develop branch.
In the product listing page => ok :heavy_check_mark: 
I tried with:
1. Attributes and their values 
image
image

2. The features and their values 
image

PS: **n** => name / **v**=> value / **u** => URL  / **pn** => public name / **me**=> meta title
image
image

Ping @PrestaShop/product-team, this PR only fix the url_name parameter in the product listing page.
But It doesn't work in the product page with url_name & meta_title as commented here:
What do you think?

> > Hi @PierreRambaud,
> > I tried to use this PR with PS1771 and with develop branch, but the same issue is displayed
> > **BO**
> > image
> > **FO**
> > image
> > 
> > * I tried with php7.1 & PHP7.2.
> > * I regenerated the .htaccess
> > * I used an incognito web
> > 
> > What do you think?
> > Thanks!
> 
> As I said here [PrestaShop/PrestaShop#16939 (comment)](https://github.com/PrestaShop/PrestaShop/issues/16939#issuecomment-752125462) this module never aims to change the product page behaviour.

Thank you!

**Comment by khouloudbelguith** [comment]
Hi @PrestaShop/product-team!

As discussed with @PierreRambaud.
This PR is ok :heavy_check_mark: we can merge it as it fixes the first problem.
The discussion has to continue on the issue: https://github.com/PrestaShop/PrestaShop/issues/16939

Thank you!

**Comment by marionf** [comment]
If only a part of the issue is fixed, we should create other(s) issue(s) with what it remains to do
And according if the fixes for each point should be done in the module or in the core

**Pull request merged**